app/_components/Poidex.tsx

```diff
@@ -0,0 +1,40 @@
+import { useState } from "react";
+import { Button } from "./ui/button";
+import PoidexModal from "./PoidexModal";
+import { Pin } from "../_utils/global";
+
+interface PoidexProps {
+  pins: Pin[]
+}
+
+const Poidex = ({pins}:PoidexProps):React.JSX.Element => {
+  const [showPoidex, setShowPoidex] = useState<boolean>(false);
+  const [selectedPoi, setSelectedPoi] = useState<Pin | null> (null);
+
+  const handleClosePoidex = () => {
+    setShowPoidex(false);
+    setSelectedPoi(null);
+  }
+
+  const handlePoiClick = (poi:Pin) => {
+    setSelectedPoi(poi);
+  }
+
+  return (
+    <>
+    <Button onClick={():void => void setShowPoidex(true)}>Open Poidex</Button>
```

Comment:
`setShowPoidex` could probably be refactored for simplicity.

```suggestion
  const handlePoidex = () => {
    setShowPoidex((prev) => !prev);
    if (selectedPoi) setSelectedPoi(null);
  }
```